psi4/src/psi4/dfocc/ccd_3index_intr.cc

```diff
@@ -36,17 +36,75 @@ namespace psi {
 namespace dfoccwave {
 
 void DFOCC::ccd_3index_intr() {
-    // defs
-    SharedTensor2d T, U;
-
-    // T(Q,ia) = \sum_{jb} b_jb^Q u_ij^ab = \sum_{jb} b(Q,jb) U(jb,ia)
-    U = SharedTensor2d(new Tensor2d("U2 (IA|JB)", naoccA, navirA, naoccA, navirA));
-    ccsd_u2_amps(U, t2);
-    T = SharedTensor2d(new Tensor2d("T2 (Q|IA)", nQ, naoccA, navirA));
-    T->gemm(false, false, bQiaA, U, 1.0, 0.0);
-    U.reset();
-    T->write(psio_, PSIF_DFOCC_AMPS);
-    T.reset();
+
+    // RHF
+    if (reference_ == "RESTRICTED") {
+        // defs
+        SharedTensor2d T, U;
```

Comment:
This comment is beyond scope, but worth including for the record: there is no good reason for `T` or `U` to be pointers rather than raw objects.